pyproject.toml

```diff
@@ -0,0 +1,24 @@
+[project]
+name = "lundpython"
+version = "0.1.0"
+description = "Add your description here"
+readme = "README.md"
+requires-python = ">=3.12"
+dependencies = [
+    "astropy>=7.0.0",
+    "jupyter>=1.1.1",
+    "jupyterlab-rise>=0.42.0",
+    "matplotlib>=3.9.2",
+    "numpy>=2.1.3",
+]
+
+[dependency-groups]
+dev = [
+    "pre-commit>=4.0.1",
```

Comment:
I wouldn't write down `pre-commit` as a dependency here. On one hand, listing it as a dependency is not good enough because that just installs the framework, but the hooks still need to be installed in a separate manual step anyways. On the other hand, there aren't any benefits to isolating `pre-commit` because the framework sets up separate environments for each of the hooks itself.